optuna/study.py

```diff
@@ -409,17 +414,62 @@ def ask(self) -> trial_module.Trial:
 
                 study.tell(trial, x ** 2)
 
+        Example:
+
+            Passing previously defined distributions to the :func:`~optuna.study.Study.ask`
+            method.
+
+            .. testcode::
+
+                import optuna
+
+
+                study = optuna.create_study()
+
+                # For example, the distributions are previously defined when using `create_trial`.
+                distributions = {
+                    "optimizer": optuna.distributions.CategoricalDistribution(["adam", "sgd"]),
+                    "lr": optuna.distributions.LogUniformDistribution(0.0001, 0.1),
+                }
+                trial = optuna.trial.create_trial(
+                    params={"optimizer": "adam", "lr": 0.0001},
+                    distributions=distributions,
+                    value=0.5,
+                )
+                study.add_trial(trial)
```

Comment:
I don't have a strong opinion, but these processes that `add_trial` to existing `study` are not directly related to `study.ask(fixed_distributions)`, so it is a bit strange to have them in the examples in this document. Since these processes are listed as concrete use cases in the PR, why not exclude them from the example?

Reply:
Thanks for your comment. The motivation is to demonstrate a scenario in the document, where the search space have to be previously defined. But, I think we can just remove `add_trial` and `create_trial` like the following snippet.

```python
import optuna

study = optuna.create_study()

distributions = {
    "optimizer": optuna.distributions.CategoricalDistribution(["adam", "sgd"]),
    "lr": optuna.distributions.LogUniformDistribution(0.0001, 0.1),
}

# You can pass the distributions previously defined.
trial = study.ask(fixed_distributions=distributions)

# `optimizer` and `lr` are already suggested and accessible with `trial.params`.
assert "optimizer" in trial.params
assert "lr" in trial.params
```

Reply:
Looks solid! It's simple and easy to understand how to use this feature.

Reply:
Thanks. I removed the lines from the example.